### PR TITLE
use try-with-resource, java.nio.file.Files#readString

### PR DESCRIPTION
**Pull request by jukzi**

### CODE REVIEW

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/APIFileGenerator.java [hunk 1]
```diff
@@ -212,39 +212,27 @@ public void generateAPIFile() {
...
 							if (isZipJarFile(currentManifest.getName())) {
-								zipFile = new ZipFile(currentManifest);
-								final ZipEntry entry = zipFile.getEntry("META-INF/MANIFEST.MF"); //$NON-NLS-1$
-								if (entry != null) {
-									inputStream = new BufferedInputStream(zipFile.getInputStream(entry));
+								try (ZipFile zipFile = new ZipFile(currentManifest)) {
+									final ZipEntry entry = zipFile.getEntry("META-INF/MANIFEST.MF"); //$NON-NLS-1$
+									if (entry != null) {
+										try (BufferedInputStream inputStream = new BufferedInputStream(
+												zipFile.getInputStream(entry))) {
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/APIFileGenerator.java:221]: `ManifestElement.parseBundleManifest` already wraps the stream passed into a `BufferedInputStream` so the wrapping can be removed here and below.

And in general Inputstreams returned by a ZipFile are closed when the ZipFile is closed, so you can skip wrapping them in a try-with-resources here and below.

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiDescriptionProcessor.java [hunk 2]
```diff
@@ -443,46 +443,29 @@ private ApiDescriptionProcessor() {
...
-					jarFile = new ZipFile(location, ZipFile.OPEN_READ);
-					ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.COMPONENT_XML_NAME);
-					if (manifestEntry != null) {
-						stream = jarFile.getInputStream(manifestEntry);
+					try (ZipFile jarFile = new ZipFile(location, ZipFile.OPEN_READ)) {
+						ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.COMPONENT_XML_NAME);
+						if (manifestEntry != null) {
+							try (InputStream stream = jarFile.getInputStream(manifestEntry)) {
+								return new String(Util.getInputStreamAsCharArray(stream, StandardCharsets.UTF_8));
+							}
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiDescriptionProcessor.java:454]: You could use `InputStream.readAllBytes()` here:
```suggestion
							byte[] allBytes = jarFile.getInputStream(manifestEntry).readAllBytes();
							return new String(allBytes, StandardCharsets.UTF_8);
```

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/SystemApiDescriptionProcessor.java [hunk 3]
```diff
@@ -65,46 +65,29 @@ private SystemApiDescriptionProcessor() {
...
-					jarFile = new ZipFile(location, ZipFile.OPEN_READ);
-					ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.SYSTEM_API_DESCRIPTION_XML_NAME);
-					if (manifestEntry != null) {
-						stream = jarFile.getInputStream(manifestEntry);
+					try (ZipFile jarFile = new ZipFile(location, ZipFile.OPEN_READ)) {
+						ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.SYSTEM_API_DESCRIPTION_XML_NAME);
+						if (manifestEntry != null) {
+							try (InputStream stream = jarFile.getInputStream(manifestEntry)) {
+								return new String(Util.getInputStreamAsCharArray(stream, StandardCharsets.UTF_8));
+							}
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/SystemApiDescriptionProcessor.java:76]:
```suggestion
							byte[] allBytes = jarFile.getInputStream(manifestEntry).readAllBytes();
							return new String(allBytes, StandardCharsets.UTF_8);
```

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/APIFileGenerator.java (hunk 1 above)
Comment (jukzi) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/APIFileGenerator.java]: done

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiDescriptionProcessor.java (hunk 2 above)
Comment (jukzi) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiDescriptionProcessor.java]: done

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/SystemApiDescriptionProcessor.java (hunk 3 above)
Comment (jukzi) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/SystemApiDescriptionProcessor.java]: done

**Comment by jukzi** [comment]
unrelated "Build: warning in 'recordIssues' step" from maven plugin
`MavenProject: org.eclipse.pde:org.eclipse.e4.tools.persistence.feature:1.1.300-SNAPSHOT @ /home/jenkins/agent/workspace/eclipse.pde_PR-1092/e4tools/features/org.eclipse.e4.tools.persistence.feature/.polyglot.feature.xml: baseline and build artifacts have same version but different contents`
image

**Pull request merged**